test_board_game_app.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+from fastapi.testclient import TestClient
+from fastapi import HTTPException
+from Board_Game_App import create_games_by_player, all_games, PlayerNotFoundError, app
+from collections import Counter
+
+### Setup Test Data ###
+em_games = {
+         "Boggle": 7,
+         "Hanabi": 6,
+         "Mysterium": 9,
+         "Rivals of Catan": 8
+      }
+
+mel_games = {
+         "Codenames": 8,
+         "Hanabi": 8,
+         "Mysterium": 7,
+         "Settlers of Catan": 6
+      }
+
+games_data = [
+    {
+        "Name": "Em",
+        "Games": em_games
+    },
+    {
+        "Name": "Mel",
+        "Games": mel_games
+    }
+]
+
+client = TestClient(app)
+
+games_by_player = create_games_by_player(games_data)
+app.games_by_player = games_by_player
+
+all_player_games = all_games(games_by_player)
+app.all_player_games = all_player_games
+
+### Test Supporting Funtions ###
+def test_create_games_by_player():
+    res = create_games_by_player(players_games_list=games_data)
+    assert res == {
+        "Em": {
+            "Boggle": 7,
+            "Hanabi": 6,
+            "Mysterium": 9,
+            "Rivals of Catan": 8
+            },
+        "Mel":{
+            "Codenames": 8,
+            "Hanabi": 8,
+            "Mysterium": 7,
+            "Settlers of Catan": 6
+            }
+        }
+    
+def test_all_games():
+    games = create_games_by_player(games_data)
+    res = all_games(games_by_player = games)
+    expected_list = ["Boggle", "Hanabi", "Mysterium", "Rivals of Catan", "Codenames", "Settlers of Catan"]
+    assert Counter(res) == Counter(expected_list)
```

Comment:
`Counter` is a cool class, but I think it's overkill in this case since there is always just one of each item. Instead you can just sort the lists:
```suggestion
    assert sorted(res) == sorted(expected_list)
```